judge/sql_query.py

```diff
@@ -141,18 +141,13 @@ def match_regex(self, regex: str) -> Union[None, str]:
 
         return None
 
-    def match_array(self, words: list[str]) -> Union[None, str]:
+    def match_array(self, words: list[str]) -> Optional[str]:
         """checks if query contains symbol that is in the list (case insensitive)
 
         :return: word that is in the list, if not found return None
         """
-        lowercase_words = [word.lower() for word in words]
-
-        for symbol in self.symbols:
-            if symbol.lower() in lowercase_words:
-                return symbol
-
-        return None
+        lowercase_words = set(map(lambda w: w.lower(), words))
+        return next((sym for sym in set(self.symbols) if sym in lowercase_words), None)
```

Comment:
These 2 line changes are probably the cause of the https://github.com/dodona-edu/judge-sql/issues/89 bug.
Maybe split this PR and change those lines in another PR @thepieterdc?
I'll take a closer look at the problem next week.

Reply:
The behaviour should be exactly the same so I would assume that there's a bug somewhere else. If you find it, please add a regression test to prevent it from happening again and to increase the coverage in general.

Reply:
The order of the elements in `self.symbols` does matter here. `match_array` should return the first match, extra unit tests can indeed verify this behavior; since this is the only functional change in this PR, the bug must be caused by these 2 lines. Will verify next week.